src/frontend/packages/cf-autoscaler/src/shared/list-types/app-autoscaler-event/cf-app-autoscaler-events-data-source.ts

```diff
@@ -30,11 +32,27 @@ export class CfAppAutoscalerEventsDataSource extends ListDataSource<APIResource<
         paginationKey: action.paginationKey,
         isLocal: false,
         listConfig,
+        // why we don't pass action and rely on metricsAction attribute?
         refresh: () => {
           if (this.metricsAction.windowValue) {
             this.metricsAction = metricsRangeService.getNewTimeWindowAction(this.metricsAction, this.metricsAction.windowValue);
           }
           this.store.dispatch(this.metricsAction);
+        },
+        customUpdate: (newAction: GetAppAutoscalerScalingHistoryAction) => {
+          const rangeParams = {
+            'start-time': newAction.query.params.start + '000000000',
+            'end-time': newAction.query.params.end + '000000000',
+          };
+
+          this.pagination$.pipe(
+            first()
```

Comment:
Do we need to use ` this.pagination$.pipe` here? It would be nice if we could dispatch a AddParam action and remove the need to get the current params from the store.